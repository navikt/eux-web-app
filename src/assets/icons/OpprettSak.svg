<svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
  <path fill-rule="evenodd" clip-rule="evenodd" d="M17 10.026V5.5C17 5.367 16.947 5.24 16.854 5.146L11.854 0.146C11.76 0.053 11.633 0 11.5 0H0.5C0.225 0 0 0.224 0 0.5V21.5C0 21.776 0.225 22 0.5 22H11.514C10.568 20.744 10 19.189 10 17.5C10 13.533 13.098 10.284 17 10.026ZM11.5 0.5L16.5 5.5H11.5V0.5ZM17.5 11C13.916 11 11 13.915 11 17.5C11 21.084 13.916 24 17.5 24C21.084 24 24 21.084 24 17.5C24 13.915 21.084 11 17.5 11ZM20.229 18H18V20.227C18 20.503 17.777 20.727 17.5 20.727C17.225 20.727 17 20.503 17 20.227V18H14.773C14.496 18 14.273 17.776 14.273 17.5C14.273 17.224 14.496 17 14.773 17H17V14.772C17 14.496 17.225 14.272 17.5 14.272C17.777 14.272 18 14.496 18 14.772V17H20.229C20.504 17 20.729 17.224 20.729 17.5C20.729 17.776 20.504 18 20.229 18Z" fill="#0067C5"/>
</svg>
