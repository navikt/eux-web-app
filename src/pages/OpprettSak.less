
.opprettsak {
  &__statuslinje {
    margin-top: 2em;
  }

  &__tema {
    display: flex;
    align-items: flex-end;

    &.feil {
      align-items: center !important;
    }
  }
  .arbeidsforhold-item {
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: space-between;

    .arbeidsforhold-desc {
      display: flex;
      flex-direction: row;
    }
  }
}


