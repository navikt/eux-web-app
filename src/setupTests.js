/* eslint-disable */

import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'
import each from 'jest-each';

global.fetch = require('jest-fetch-mock');

Enzyme.configure({ adapter: new Adapter() });

global.shallow = shallow;
global.mount = mount;

global.each = each;
