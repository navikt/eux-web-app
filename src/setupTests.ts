import {render, screen} from '@testing-library/react'
import { act } from 'react-dom/test-utils'
import { useDispatch, useSelector } from 'react-redux'
import 'jest-styled-components'
import '@testing-library/jest-dom/extend-expect'

jest.mock('react-redux');

(global as any).screen = screen;
(global as any).render = render;
(global as any).act = act

// eslint-disable-next-line no-undef
HTMLCanvasElement.prototype.getContext = jest.fn()
window.scrollTo = jest.fn()

Object.defineProperty(window, 'getComputedStyle', {
  value: () => ({
    getPropertyValue: () => ('')
  })
})

jest.mock('amplitude-js', () => ({
  getInstance: () => ({
    init: jest.fn(),
    logEvent: jest.fn()
  })
}))

jest.mock('i18next', () => {
  const use = jest.fn()
  const init = jest.fn()
  const loadLanguages = jest.fn()
  const result = {
    use: use,
    init: init,
    loadLanguages: loadLanguages
  }
  use.mockImplementation(() => result)
  return result
})

jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: (key: any) => key })
}))

export const stageSelector = (defaultSelector: any, params: any) => {
  (useDispatch as jest.Mock).mockImplementation(() => jest.fn());
  (useSelector as jest.Mock).mockImplementation(() => ({
    ...defaultSelector,
    ...params
  }))
}
