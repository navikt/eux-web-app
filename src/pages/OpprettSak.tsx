import * as appActions from 'actions/app'
import * as formActions from 'actions/form'
import * as sakActions from 'actions/sak'
import AbortModal from 'components/AbortModal/AbortModal'
import Family from 'components/Family/Family'
import PersonSearch from 'components/PersonSearch/PersonSearch'
import TopContainer from 'components/TopContainer/TopContainer'
import { State } from 'declarations/reducers'
import { Enheter, FagSaker, Person, Validation } from 'declarations/types'
import * as EKV from 'eessi-kodeverk'
import Ui from 'eessi-pensjon-ui'
import _ from 'lodash'
import PT from 'prop-types'
import React, { useState } from 'react'
import { useTranslation } from 'react-i18next'
import { useDispatch, useSelector } from 'react-redux'
import { Link } from 'react-router-dom'
import IkonArbeidsforhold from 'resources/images/ikon-arbeidsforhold'
import { formatterDatoTilNorsk } from 'utils/dato'
import classNames from 'classnames'
import './OpprettSak.css'

export interface OpprettSakProps {
  history: any;
}

export interface OpprettSakSelector {
  arbeidsforhold: any;
  buctyper: any;
  fagsaker: FagSaker | undefined | null;
  institusjoner: any;
  kodemaps: any;
  landkoder: any;
  opprettetSak: any;
  sektor: any;
  enheter: Enheter | undefined;
  person: Person | undefined;
  sedtyper: any;
  sendingSak: boolean;
  serverInfo: any;
  tema: any;

  valgtFnr: any;
  valgtUnit: any;
  valgtBucType: any;
  valgtSedType: any;
  valgtSektor: any;
  valgtLandkode: any;
  valgtInstitusjon: any;
  valgtSaksId: any;
  valgtTema: any,
  valgteFamilieRelasjoner: any;
  valgteArbeidsforhold: any;
}

const mapState = (state: State): OpprettSakSelector => ({
  enheter: state.app.enheter,
  serverInfo: state.app.serverinfo,

  sendingSak: state.loading.sendingSak,

  arbeidsforhold: state.sak.arbeidsforhold,
  buctyper: state.sak.buctyper,
  fagsaker: state.sak.fagsaker,
  kodemaps: state.sak.kodemaps,
  institusjoner: state.sak.institusjoner,
  landkoder: state.sak.landkoder,
  opprettetSak: state.sak.opprettetSak,
  person: state.sak.person,
  sedtyper: state.sak.sedtyper,
  sektor: state.sak.sektor,
  tema: state.sak.tema,

  valgteArbeidsforhold: state.form.arbeidsforhold,
  valgtBucType: state.form.buctype,
  valgteFamilieRelasjoner: state.form.familierelasjoner,
  valgtFnr: state.form.fnr,
  valgtInstitusjon: state.form.institusjon,
  valgtLandkode: state.form.landkode,
  valgtSaksId: state.form.saksId,
  valgtSedType: state.form.sedtype,
  valgtSektor: state.form.sektor,
  valgtTema: state.form.tema,
  valgtUnit: state.form.unit
})

const OpprettSak: React.FC<OpprettSakProps> = ({ history } : OpprettSakProps): JSX.Element => {
  const {
    enheter, serverInfo, sendingSak, arbeidsforhold, buctyper, fagsaker, sedtyper, institusjoner, kodemaps,
    landkoder, opprettetSak, person, sektor, tema, valgteArbeidsforhold, valgtBucType, valgteFamilieRelasjoner,
    valgtFnr, valgtInstitusjon, valgtLandkode, valgtSaksId, valgtSedType, valgtSektor, valgtTema, valgtUnit
  }: OpprettSakSelector = useSelector<State, OpprettSakSelector>(mapState)
  const dispatch = useDispatch()
  const { t } = useTranslation()

  const [visModal, setVisModal] = useState(false)
  const [validation, setValidation] = useState<{[k: string]: any}>({})
  const [isFnrValid, setIsFnrValid] = useState<boolean>(false)

  const temaer = (!kodemaps ? [] : (!valgtSektor ? [] : tema[kodemaps.SEKTOR2FAGSAK[valgtSektor]]))
  const _buctyper = (!kodemaps ? [] : (!valgtSektor ? [] : buctyper[kodemaps.SEKTOR2FAGSAK[valgtSektor]]))
  let _sedtyper = (!kodemaps ? [] : ((!valgtSektor || !valgtBucType) ? [] : kodemaps.BUC2SEDS[valgtSektor][valgtBucType]))

  if (!(_sedtyper && _sedtyper.length)) {
    _sedtyper = []
  }
  _sedtyper = _sedtyper.reduce((acc: any, curr: any) => {
    const kode = sedtyper?.find((elem: any) => elem.kode === curr)
    acc.push(kode)
    return acc
  }, [])

  const isSomething = (value: any): boolean => (!_.isNil(value) && !_.isEmpty(value))
  const visFagsakerListe: boolean = isSomething(valgtSektor) && isSomething(tema) && isSomething(fagsaker)
  const visArbeidsforhold: boolean = EKV.Koder.sektor.FB === valgtSektor && EKV.Koder.buctyper.family.FB_BUC_01 === valgtBucType && isSomething(valgtSedType)

  const validate = (): Validation => {
    const validation: Validation = {
      fnr: !valgtFnr ? t('ui:validation-noFnr') : (!isFnrValid ? t('ui:validation-uncheckedFnr') : null),
      sektor: !valgtSektor ? t('ui:validation-noSektor') : null,
      buctype: !valgtBucType ? t('ui:validation-noBuctype') : null,
      sedtype: !valgtSedType ? t('ui:validation-noSedtype') : null,
      landkode: !valgtLandkode ? t('ui:validation-noLand') : null,
      institusjon: !valgtInstitusjon ? t('ui:validation-noInstitusjonsID') : null,
      tema: !valgtTema ? t('ui:validation-noTema') : null,
      saksId: !valgtSaksId ? t('ui:validation-noSaksId') : null
    }
    setValidation(validation)
    return validation
  }

  const resetAllValidation = () => {
    setValidation({})
  }

  const resetValidation = (key: Array<string> | string): void => {
    const newValidation = _.cloneDeep(validation)
    if (_.isString(key)) {
      newValidation[key] = null
    }
    if (_.isArray(key)) {
      key.forEach(k => {
        newValidation[k] = null
      })
    }
    setValidation(newValidation)
  }

  const isValid = (_validation: Validation): boolean => {
    return _.find(_.values(_validation), e => e !== null) === undefined
  }

  const skjemaSubmit = (): void => {
    if (isValid(validate())) {
      dispatch(sakActions.createSak({
        buctype: valgtBucType,
        fnr: valgtFnr,
        landKode: valgtLandkode,
        institusjonsID: valgtInstitusjon,
        saksID: valgtSaksId,
        sedtype: valgtSedType,
        sektor: valgtSektor,
        tema: valgtTema,
        familierelasjoner: valgteFamilieRelasjoner,
        arbeidsforhold: valgteArbeidsforhold
      }))
    }
  }

  const openModal = (): void => {
    setVisModal(true)
  }

  const closeModal = (): void => {
    setVisModal(false)
  }

  const onAbort = (): void => {
    dispatch(appActions.cleanData())
    history.push('/')
  }

  const onUnitChange = (e: React.ChangeEvent<HTMLSelectElement>): void => {
    resetValidation('unit')
    dispatch(formActions.set('unit', e.target.value))
  }

  const onSektorChange = (e: React.ChangeEvent<HTMLSelectElement>): void => {
    resetValidation('sektor')
    resetValidation('unit')
    dispatch(formActions.set('unit', undefined))
    dispatch(formActions.set('sektor', e.target.value))
  }

  const onBuctypeChange = (event: React.ChangeEvent<HTMLSelectElement>): void => {
    resetValidation(['buctype', 'landkode'])
    const buctype = event.target.value
    dispatch(formActions.set('buctype', buctype))
    dispatch(formActions.set('landkode', undefined))
    dispatch(formActions.set('sedtype', undefined))
    dispatch(formActions.set('institution', undefined))
    dispatch(sakActions.getLandkoder(buctype))
  }

  const onSedtypeChange = (e: React.ChangeEvent<HTMLSelectElement>): void => {
    resetValidation('sedtype')
    dispatch(formActions.set('sedtype', e.target.value))
  }

  const onSedtypeSet = (e: string): void => {
    resetValidation('sedtype')
    dispatch(formActions.set('sedtype', e))
  }

  const onLandkodeChange = (country: any): void => {
    resetValidation(['landkode', 'institusjon'])
    const landKode = country.value
    dispatch(formActions.set('landkode', landKode))
    dispatch(sakActions.getInstitusjoner(valgtBucType, landKode))
  }

  const onInstitusjonChange = (event: React.ChangeEvent<HTMLSelectElement>): void => {
    resetValidation('institusjon')
    dispatch(formActions.set('institusjon', event.target.value))
  }

  const onTemaChange = (event: React.ChangeEvent<HTMLSelectElement>): void => {
    resetValidation(['tema', 'saksId'])
    dispatch(formActions.set('tema', event.target.value))
    dispatch(formActions.set('fagsaker', undefined))
  }

  const onViewFagsakerClick = (): void => {
    dispatch(sakActions.getFagsaker(person?.fnr, valgtSektor, valgtTema))
  }

  const onSakIDChange = (event: React.ChangeEvent<HTMLSelectElement>): void => {
    resetValidation('saksId')
    dispatch(formActions.set('saksId', event.target.value))
  }

  const getArbeidsforhold = (): void => {
    dispatch(sakActions.getArbeidsforhold(person?.fnr))
  }

  const onArbeidsforholdClick = (item: any, checked: boolean): void => {
    if (checked) {
      dispatch(formActions.addArbeidsforhold(item))
    } else {
      dispatch(formActions.removeArbeidsforhold(item))
    }
  }

  return (
    <TopContainer className='opprettsak'>
      <Ui.Nav.Row className='m-0 p-4'>
        <div className='col-sm-1' />
        <div className='col-sm-10'>
          <Ui.Nav.Systemtittel className='mb-4'>
            {t('ui:title-newcase')}
          </Ui.Nav.Systemtittel>
          <PersonSearch
            className='slideAnimate'
            validation={validation}
            resetAllValidation={resetAllValidation}
            onFnrChange={() => setIsFnrValid(false)}
            onPersonFound={() => setIsFnrValid(true)}
          />
          {person ? (
            <Ui.Nav.Row>
              <div className='col-xs-6 slideAnimate' style={{ animationDelay: '0s' }}>
                <Ui.Nav.Select
                  className='mb-4'
                  id='id-sektor'
                  label={t('ui:label-sektor')}
                  disabled={!person}
                  onChange={onSektorChange}
                  value={valgtSektor}
                  feil={validation.sektor}
                >
                  <option value=''>{t('ui:form-choose')}</option>)
                  {sektor ? _.orderBy(sektor, 'term').map((element: any) => (
                    <option value={element.kode} key={element.kode}>{element.term}</option>)
                  ) : null}
                </Ui.Nav.Select>
              </div>
              <div className='col-xs-6 slideAnimate' style={{ animationDelay: '0.15s' }}>
                <Ui.Nav.Select
                  className='mb-4'
                  id='id-enhet'
                  disabled={!(valgtSektor === 'HZ' || valgtSektor === 'SI')}
                  label={t('ui:label-unit')}
                  onChange={onUnitChange}
                  value={valgtUnit}
                  feil={validation.unit}
                >
                  <option value=''>{t('ui:form-choose')}</option>)
                  {sektor ? _.orderBy(enheter, 'navn').map((element: any) => (
                    <option value={element.enhetId} key={element.enhetId}>{element.navn}</option>)
                  ) : null}
                </Ui.Nav.Select>
              </div>
              <div className='col-xs-6 slideAnimate' style={{ animationDelay: '0.15s' }}>
                <Ui.Nav.Select
                  className='mb-4'
                  id='id-buctype'
                  label={t('ui:label-buc')}
                  disabled={!isSomething(valgtSektor)}
                  onChange={onBuctypeChange}
                  value={valgtBucType}
                  feil={validation.buctype}
                >
                  <option value=''>{t('ui:form-choose')}</option>)
                  {_buctyper ? _.orderBy(_buctyper, 'kode').map((element: any) => (
                    <option value={element.kode} key={element.kode}>{element.kode} - {element.term}</option>)
                  ) : null}
                </Ui.Nav.Select>
              </div>
              <div className='col-xs-6 slideAnimate' style={{ animationDelay: '0.3s' }}>
                <Ui.Nav.Select
                  className='mb-4'
                  id='id-sedtype'
                  label={t('ui:label-sed')}
                  disabled={!isSomething(valgtBucType) || !isSomething(valgtSektor)}
                  onChange={onSedtypeChange}
                  value={valgtSedType}
                  feil={validation.sedtype}
                >
                  <option value=''>{t('ui:form-choose')}</option>)
                  {_sedtyper ? _sedtyper.map((element: any) => {
                    // if only one element, select it
                    if (_sedtyper.length === 1 && valgtSedType !== element.kode) {
                      onSedtypeSet(element.kode)
                    }
                    return <option value={element.kode} key={element.kode}>{element.kode} - {element.term}</option>
                  }) : null}
                </Ui.Nav.Select>
              </div>
              <div className='col-xs-6 slideAnimate' style={{ animationDelay: '0.45s' }}>
                <Ui.CountrySelect
                  className='mb-4'
                  label={t('ui:label-landkode')}
                  lang='nb'
                  placeholder={t('ui:form-choose')}
                  menuPortalTarget={document.body}
                  disabled={!isSomething(person)}
                  includeList={landkoder ? _.orderBy(landkoder, 'term').map((element: any) => element.kode) : []}
                  onOptionSelected={onLandkodeChange}
                  value={valgtLandkode}
                  error={validation.landkode}
                />
              </div>
              <div className='col-xs-6 slideAnimate' style={{ animationDelay: '0.6s' }}>
                <Ui.Nav.Select
                  id='id-institusjon'
                  className='mb-4'
                  disabled={!isSomething(valgtLandkode)}
                  value={valgtInstitusjon}
                  onChange={onInstitusjonChange}
                  label={t('ui:label-institusjon')}
                  feil={validation.institusjon}
                >
                  <option value=''>{t('ui:form-choose')}</option>)
                  {institusjoner ? _.orderBy(institusjoner, 'term').map((element: any) => (
                    <option value={element.institusjonsID} key={element.institusjonsID}>{element.navn}</option>)
                  ) : null}
                </Ui.Nav.Select>
              </div>
              {valgtSektor === 'FB' ? (
                <div className='col-xs-12 mb-4 slideAnimate'>
                  <Family />
                </div>
              ) : null}
              {valgtSektor ? (
                <>
                  <div className='col-xs-12'>
                    <div className={classNames('slideAnimate', 'opprettsak__tema', { feil: !!validation.tema })}>
                      <div className='w-50 mr-3'>
                        <Ui.Nav.Select
                          id='id-behandlings-tema'
                          className='mb-4'
                          label={t('ui:label-tema')}
                          value={valgtTema}
                          onChange={onTemaChange}
                          feil={validation.tema}
                        >
                          <option value=''>{t('ui:form-choose')}</option>)
                          {temaer ? temaer.map((element: any) => (
                            <option value={element.kode} key={element.kode}>{element.term}</option>)
                          ) : null}
                        </Ui.Nav.Select>
                      </div>
                      <div className='w-50'>
                        <div className='d-flex' style={{ alignItems: 'center', justifyContent: 'space-between' }}>
                          <div>
                            <Ui.Nav.Knapp
                              className='mb-4'
                              onClick={onViewFagsakerClick}
                              disabled={!isSomething(valgtTema)}
                            >
                              {t('ui:form-seeCases')}
                            </Ui.Nav.Knapp>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  {fagsaker === null || (fagsaker !== undefined && _.isEmpty(fagsaker)) ? (
                    <div className='col-xs-12 mb-4'>
                      <Ui.Nav.AlertStripe type='advarsel'>
                        {t('ui:error-fagsak-notFound')}
                        <Ui.Nav.Lenke
                          className='mb-4'
                          href={serverInfo.gosysURL}
                          ariaLabel={t('ui:form-createNewCaseInGosys')}
                          target='_blank'
                        >
                          {t('ui:form-createNewCaseInGosys')}
                        </Ui.Nav.Lenke>
                      </Ui.Nav.AlertStripe>
                    </div>
                  ) : null}
                </>
              ) : null}
              {visFagsakerListe ? (
                <>
                  <div className='col-xs-6'>
                    <Ui.Nav.Select
                      id='id-fagsaker'
                      className='mb-4'
                      label={t('ui:label-fagsak')}
                      value={valgtSaksId}
                      onChange={onSakIDChange}
                      feil={validation.saksId}
                    >
                      <option value=''>{t('ui:form-choose')}</option>
                      {fagsaker ? _.orderBy(fagsaker, 'fagsakNr').map(element => (
                        <option value={element.saksID} key={element.saksID}>{element.fagsakNr ? element.fagsakNr : element.saksID}</option>)
                      ) : null}
                    </Ui.Nav.Select>
                  </div>
                  <div className='col-xs-6' />
                </>
              ) : null}
              {visArbeidsforhold ? (
                <>
                  <div className='col-xs-6 arbeidsforhold'>
                    <Ui.Nav.Row>
                      <div className='col-xs-6'>
                        <strong>{t('ui:label-aaRegistered')}</strong><br />{t('ui:label-arbeidsforhold')}
                      </div>
                      <div className='col-xs-6'>
                        <Ui.Nav.Knapp onClick={getArbeidsforhold}>
                          {t('ui:form-search')}
                        </Ui.Nav.Knapp>
                      </div>
                    </Ui.Nav.Row>
                    {arbeidsforhold ? arbeidsforhold.map((arbeidsforholdet: any, index: number) => {
                      const { arbeidsforholdIDnav, navn, orgnr, ansettelsesPeriode: { fom, tom } } = arbeidsforholdet
                      const arbeidsForholdErValgt = valgteArbeidsforhold.find((item: any) => item.arbeidsforholdIDnav === arbeidsforholdIDnav)
                      return (
                        <Ui.Nav.Panel key={index} className='mt-4' border>
                          <div className='arbeidsforhold-item'>
                            <div className='arbeidsforhold-desc'>
                              <div className='mr-3'>
                                <IkonArbeidsforhold />
                              </div>
                              <div>
                                <strong>{navn}</strong>
                                <br />
                                {t('ui:label-orgnummer')}:&nbsp;{orgnr}
                                <br />
                                {t('ui:label-startDate')}:&nbsp;{formatterDatoTilNorsk(fom)}
                                <br />
                                {t('ui:label-endDate')}:&nbsp;{formatterDatoTilNorsk(tom)}
                              </div>
                            </div>
                            <div>
                              <Ui.Nav.Checkbox
                                checked={arbeidsForholdErValgt}
                                onChange={(e: React.ChangeEvent<HTMLInputElement>) => onArbeidsforholdClick(arbeidsforholdet, e.target.checked)}
                                label={t('ui:form-choose')}
                              />
                            </div>
                          </div>
                        </Ui.Nav.Panel>
                      )
                    }) : null}
                  </div>
                  <div className='col-xs-6' />
                </>
              ) : null}
              <div className='col-xs-12 buttons mt-4 slideAnimate' style={{ animationDelay: '0.75s' }}>
                <Ui.Nav.Hovedknapp
                  className='mr-4'
                  disabled={sendingSak}
                  onClick={skjemaSubmit}
                  spinner={sendingSak}
                >
                  {t('ui:form-createCaseInRina')}
                </Ui.Nav.Hovedknapp>
                <Ui.Nav.Flatknapp aria-label='Navigasjonslink tilbake til forsiden' onClick={openModal}>
                  {t('ui:form-resetForm')}
                </Ui.Nav.Flatknapp>
              </div>
              {opprettetSak && opprettetSak.url ? (
                <div className='col-xs-12'>
                  <Ui.Nav.AlertStripe className='mt-4 w-50' type='suksess'>
                    <div>
                      {opprettetSak.rinasaksnummer ? (
                        <Link to={'/vedlegg?rinasaksnummer=' + opprettetSak.rinasaksnummer}>
                          {t('ui:form-caseNumber') + ': ' + opprettetSak.rinasaksnummer}
                        </Link>
                      ) : <span>{t('ui:form-caseNumber') + ': ' + opprettetSak.rinasaksnummer}</span>}

                      <span className='ml-1 mr-1'>{t('ui:label-is-created')}.</span>
                      {opprettetSak.url ? (
                        <Ui.Nav.Lenke className='vedlegg__lenke ml-1 mr-1' href={opprettetSak.url} target='_blank'>
                          {t('ui:form-goToRina')}
                        </Ui.Nav.Lenke>
                      ) : null}
                    </div>
                  </Ui.Nav.AlertStripe>
                </div>
              ) : null}
            </Ui.Nav.Row>
          ) : null}
          <AbortModal
            onAbort={onAbort}
            isOpen={visModal}
            closeModal={closeModal}
          />
        </div>
        <div className='col-sm-1' />
      </Ui.Nav.Row>
    </TopContainer>
  )
}

OpprettSak.propTypes = {
  history: PT.any.isRequired
}

export default OpprettSak
