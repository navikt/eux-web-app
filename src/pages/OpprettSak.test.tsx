import { mount, ReactWrapper } from 'enzyme'
import React from 'react'
import { stageSelector } from 'setupTests'
import OpprettSak, { OpprettSakProps, OpprettSakSelector } from './OpprettSak'

jest.mock('react-router-dom', () => ({
  Link: () => (<div className='mock-link' />)
}))

describe('pages/OpprettSak/OpprettSak', () => {
  let wrapper: ReactWrapper
  const initialMockProps: OpprettSakProps = {
    history: {}
  }

  const defaultSelector: OpprettSakSelector = {
    alertMessage: undefined,
    alertStatus: undefined,
    alertType: undefined,

    enheter: undefined,
    serverInfo: undefined,
    buctyper: undefined,
    familierelasjonKodeverk: undefined,
    kodemaps: undefined,
    landkoder: undefined,
    sedtyper: undefined,
    sektor: undefined,
    tema: undefined,

    sendingSak: false,
    gettingPerson: false,

    arbeidsperioder: undefined,
    valgteArbeidsgivere: [],
    valgtBucType: undefined,
    valgteFamilieRelasjoner: [],
    fagsaker: undefined,
    valgtFnr: undefined,
    valgtInstitusjon: undefined,
    institusjoner: undefined,
    valgtLandkode: undefined,
    opprettetSak: undefined,
    person: undefined,
    personRelatert: undefined,
    valgtSaksId: undefined,
    valgtSedType: undefined,
    valgtSektor: undefined,
    valgtTema: undefined,
    valgtUnit: undefined
  }

  beforeAll(() => {
    stageSelector(defaultSelector, {})
  })

  beforeEach(() => {
    wrapper = mount(<OpprettSak {...initialMockProps} />)
  })

  afterEach(() => {
    wrapper.unmount()
  })

  it('Renders', () => {
    expect(wrapper.isEmptyRender()).toBeFalsy()
    expect(wrapper).toMatchSnapshot()
  })

  it('HTML structure', () => {
    expect(wrapper.exists('.opprettsak')).toBeTruthy()
  })
})
