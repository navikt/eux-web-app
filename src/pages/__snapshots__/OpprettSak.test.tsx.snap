// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/OpprettSak/OpprettSak Renders 1`] = `
<OpprettSak
  history={Object {}}
>
  <TopContainer
    className="opprettsak"
  >
    <UseErrorBoundaryWrapper
      renderError={[Function]}
    >
      <ErrorBoundary
        onDidCatch={[Function]}
        renderError={[Function]}
      >
        <Header
          className=""
        >
          <header
            className="c-header"
          >
            <div
              className="c-header__brand"
            >
              <Link
                className="ml-2 mr-2"
                onClick={[Function]}
                to="/"
              >
                <div
                  className="mock-link"
                />
              </Link>
              <div
                className="c-header__skillelinje"
              />
              <div
                className="c-header__tittel"
              >
                <span>
                  ui:app-name
                </span>
              </div>
            </div>
            <Undertittel>
              <n
                type="undertittel"
              >
                <h2
                  className="typo-undertittel"
                >
                  ui:app-title
                </h2>
              </n>
            </Undertittel>
            <div
              className="c-header__saksbehandler"
            >
              <Lenke
                className="c-header__highcontrast-link mr-3"
                href="#highContrast"
                onClick={[Function]}
              >
                <a
                  className="lenke c-header__highcontrast-link mr-3"
                  href="#highContrast"
                  onClick={[Function]}
                >
                  ui:label-highContrast
                </a>
              </Lenke>
            </div>
          </header>
        </Header>
        <Alert
          onClose={[Function]}
          type="server"
        />
        <main
          className="opprettsak _container p-0 container-fluid"
          id="main"
          role="main"
        >
          <Row
            className="m-0 p-4"
          >
            <div
              className="row m-0 p-4"
            >
              <div
                className="col-sm-1"
              />
              <div
                className="col-sm-10"
              >
                <Systemtittel
                  className="mb-4"
                >
                  <n
                    className="mb-4"
                    type="systemtittel"
                  >
                    <h2
                      className="typo-systemtittel mb-4"
                    >
                      ui:title-newcase
                    </h2>
                  </n>
                </Systemtittel>
                <PersonSearch
                  className="slideAnimate"
                  onFnrChange={[Function]}
                  onPersonFound={[Function]}
                  resetAllValidation={[Function]}
                  validation={Object {}}
                >
                  <div
                    className="personsok slideAnimate"
                  >
                    <div
                      className="personsok__skjema"
                    >
                      <Input
                        bredde="fullbredde"
                        className="personsok__input"
                        id="personsok__input-id"
                        label="ui:form-searchUser"
                        onChange={[Function]}
                        value=""
                      >
                        <div
                          className="skjemaelement personsok__input"
                        >
                          <t
                            htmlFor="personsok__input-id"
                          >
                            <label
                              className="skjemaelement__label"
                              htmlFor="personsok__input-id"
                            >
                              ui:form-searchUser
                            </label>
                          </t>
                          <input
                            aria-invalid={false}
                            className="skjemaelement__input input--fullbredde"
                            id="personsok__input-id"
                            onChange={[Function]}
                            type="text"
                            value=""
                          />
                        </div>
                      </Input>
                      <Knapp
                        className="personsok__knapp"
                        onClick={[Function]}
                      >
                        <KnappBase
                          autoDisableVedSpinner={false}
                          className="personsok__knapp"
                          disabled={false}
                          htmlType="submit"
                          inaktivKlasseVedDisabled={false}
                          kompakt={false}
                          mini={false}
                          onClick={[Function]}
                          spinner={false}
                          type="standard"
                        >
                          <button
                            className="knapp personsok__knapp knapp--standard"
                            disabled={false}
                            onClick={[Function]}
                            type="submit"
                          >
                            ui:form-search
                          </button>
                        </KnappBase>
                      </Knapp>
                    </div>
                  </div>
                </PersonSearch>
                <AbortModal
                  closeModal={[Function]}
                  isOpen={false}
                  onAbort={[Function]}
                >
                  <Modal
                    ariaHideApp={false}
                    closeButton={false}
                    closeTimeoutMS={0}
                    contentLabel="ui:modal-abort-contentLabel"
                    isOpen={false}
                    onRequestClose={[Function]}
                    shouldCloseOnOverlayClick={true}
                  >
                    <t
                      ariaHideApp={false}
                      bodyOpenClassName="ReactModal__Body--open"
                      className="modal"
                      closeTimeoutMS={0}
                      contentLabel="ui:modal-abort-contentLabel"
                      isOpen={false}
                      onRequestClose={[Function]}
                      overlayClassName="modal__overlay"
                      parentSelector={[Function]}
                      portalClassName="ReactModalPortal"
                      role="dialog"
                      shouldCloseOnEsc={true}
                      shouldCloseOnOverlayClick={true}
                      shouldFocusAfterRender={true}
                      shouldReturnFocusAfterClose={true}
                    >
                      <Portal
                        containerInfo={
                          <div
                            class="ReactModalPortal"
                          />
                        }
                      >
                        <t
                          ariaHideApp={false}
                          bodyOpenClassName="ReactModal__Body--open"
                          className="modal"
                          closeTimeoutMS={0}
                          contentLabel="ui:modal-abort-contentLabel"
                          defaultStyles={
                            Object {
                              "content": Object {
                                "WebkitOverflowScrolling": "touch",
                                "background": "#fff",
                                "border": "1px solid #ccc",
                                "borderRadius": "4px",
                                "bottom": "40px",
                                "left": "40px",
                                "outline": "none",
                                "overflow": "auto",
                                "padding": "20px",
                                "position": "absolute",
                                "right": "40px",
                                "top": "40px",
                              },
                              "overlay": Object {
                                "backgroundColor": "rgba(255, 255, 255, 0.75)",
                                "bottom": 0,
                                "left": 0,
                                "position": "fixed",
                                "right": 0,
                                "top": 0,
                              },
                            }
                          }
                          isOpen={false}
                          onRequestClose={[Function]}
                          overlayClassName="modal__overlay"
                          parentSelector={[Function]}
                          portalClassName="ReactModalPortal"
                          role="dialog"
                          shouldCloseOnEsc={true}
                          shouldCloseOnOverlayClick={true}
                          shouldFocusAfterRender={true}
                          shouldReturnFocusAfterClose={true}
                          style={
                            Object {
                              "content": Object {},
                              "overlay": Object {},
                            }
                          }
                        />
                      </Portal>
                    </t>
                  </Modal>
                </AbortModal>
              </div>
              <div
                className="col-sm-1"
              />
            </div>
          </Row>
        </main>
        <SessionMonitor>
          <styled.div>
            <div
              className="sc-AxjAm kmXdIo"
            />
          </styled.div>
        </SessionMonitor>
        <Version />
      </ErrorBoundary>
    </UseErrorBoundaryWrapper>
  </TopContainer>
</OpprettSak>
`;
